src/c/tb_client_header.zig

```diff
@@ -0,0 +1,213 @@
+const std = @import("std");
+const tb = @import("../tigerbeetle.zig");
+const tb_client = @import("tb_client.zig");
+
+const type_mappings = .{
+    .{ tb.AccountFlags, "TB_ACCOUNT_FLAGS" },
+    .{ tb.Account, "tb_account_t" },
+    .{ tb.TransferFlags, "TB_TRANSFER_FLAGS" },
+    .{ tb.Transfer, "tb_transfer_t" },
+    .{ tb.CreateAccountResult, "TB_CREATE_ACCOUNT_RESULT" },
+    .{ tb.CreateTransferResult, "TB_CREATE_TRANSFER_RESULT" },
+    .{ tb.CreateAccountsResult, "tb_create_accounts_result_t" },
+    .{ tb.CreateTransfersResult, "tb_create_transfers_result_t" },
+    .{ tb_client.tb_operation_t, "TB_OPERATION" },
+    .{ tb_client.tb_packet_status_t, "TB_PACKET_STATUS" },
+    .{ tb_client.tb_packet_t, "tb_packet_t" },
+    .{ tb_client.tb_packet_list_t, "tb_packet_list_t" },
+    .{ tb_client.tb_client_t, "tb_client_t" },
+    .{ tb_client.tb_status_t, "TB_STATUS" },
+};
+
+fn resolve_c_type(comptime Type: type) []const u8 {
+    switch (@typeInfo(Type)) {
+        .Array => |info| return resolve_c_type(info.child),
+        .Enum => |info| return resolve_c_type(info.tag_type),
+        .Struct => return resolve_c_type(std.meta.Int(.unsigned, @sizeOf(Type) * 8)),
+        .Int => |info| {
+            std.debug.assert(info.signedness == .unsigned);
+            return switch (info.bits) {
+                8 => "uint8_t",
+                16 => "uint16_t",
+                32 => "uint32_t",
+                64 => "uint64_t",
+                128 => "tb_uint128_t",
+                else => @compileError("invalid int type"),
+            };
+        },
+        .Optional => |info| switch (@typeInfo(info.child)) {
+            .Pointer => return resolve_c_type(info.child),
+            else => @compileError("Unsupported optional type: " ++ @typeName(Type)),
+        },
+        .Pointer => |info| {
+            std.debug.assert(info.size != .Slice);
+            std.debug.assert(!info.is_allowzero);
+
+            inline for (type_mappings) |type_mapping| {
+                const ZigType = type_mapping[0];
+                const c_name = type_mapping[1];
+
+                if (info.child == ZigType) {
+                    const prefix = if (@typeInfo(ZigType) == .Struct) "struct " else "";
+                    return prefix ++ c_name ++ "*";
+                }
+            }
+
+            return resolve_c_type(info.child) ++ "*";
+        },
+        .Void, .Opaque => return "void",
+        else => @compileError("Unhandled type: " ++ @typeName(Type)),
+    }
+}
+
+fn to_uppercase(comptime input: []const u8) []const u8 {
+    comptime var output: [input.len]u8 = undefined;
+    inline for (output) |*char, i| {
+        char.* = input[i];
+        char.* -= 32 * @as(u8, @boolToInt(char.* >= 'a' and char.* <= 'z'));
+    }
+    return &output;
```

Comment:
I was referring to declaring an array and then returning a reference/slice to it from within the function that declares it. In a non-comptime setting that wouldn't be legal (since the array is on the stack). But in comptime I guess it works!